tests/test_cfg/test_z3_constraints.py

```diff
@@ -0,0 +1,319 @@
+import astroid
+import z3
+
+from python_ta.cfg import CFGVisitor, ControlFlowGraph
+from python_ta.transforms.z3_visitor import Z3Visitor
+
+
+def test_simple_function() -> None:
+    src = """
+    def func(x: int, y: int) -> int:
+        '''
+        Preconditions:
+            - x > 0 and y > 0
+            - x >= y
+        '''
+        print(x + y)
+    """
+    cfg = _create_cfg(src, "func")
+    x = z3.Int("x")
+    y = z3.Int("y")
+    expected = {0: [z3.And(x > 0, y > 0), x >= y]}
+    assert all(edge.z3_constraints == expected for edge in cfg.get_edges())
+
+
+def test_if_statement() -> None:
+    src = """
+    def func(x: int, y: bool) -> None:
+        '''
+        Preconditions:
+            - x > 0
+        '''
+        print("before if")
+        if x > 5 and y:
+            print("inside if")
+        print("after if")
+    """
+    cfg = _create_cfg(src, "func")
+    x = z3.Int("x")
+    y = z3.Bool("y")
+    expected_if_path = [
+        [x > 0],
+        [x > 0, z3.And(x > 5, y)],
+        [x > 0, z3.And(x > 5, y)],
+        [x > 0, z3.And(x > 5, y)],
+    ]
+    expected_other_path = [
+        [x > 0],
+        [x > 0, z3.Not(z3.And(x > 5, y))],
+        [x > 0, z3.Not(z3.And(x > 5, y))],
+    ]
+
+    actual_path_first = []
+    actual_path_second = []
+    for edge in cfg.get_edges():
+        actual1 = edge.z3_constraints.get(0)
+        actual2 = edge.z3_constraints.get(1)
+        if actual1 is not None:
+            actual_path_first.append(actual1)
+        if actual2 is not None:
+            actual_path_second.append(actual2)
+
+    assert (
+        set(actual) == set(expected)
+        for actual, expected in zip(actual_path_first, expected_if_path)
```

Comment:
`zip` alone isn't appropriate because it'll iterate until reaching the end of the _shorter_ input. You can first use a `len` check to ensure the two lists have the same length.